docs/source/tutorial.rst

```diff
@@ -0,0 +1,553 @@
+
+.. code:: 
+
+    !pip install git+https://github.com/fsspec/kerchunk
+
+Here we are considering Netcdf4 files and so use the kerchunk ``hdf`` module. Support for ``fits``, ``grib2``, ``tiff``, ``netCDF3`` and ``zarr`` are available in other kerchunk modules. Alternatively it is also possible to manually create reference jsons for more specific cases. The Earth Big Data `example <https://github.com/fsspec/kerchunk/blob/main/examples/earthbigdata.ipynb>`__ provides a demonstration of this.
+
+.. code:: 
+
+    from kerchunk.hdf import SingleHdf5ToZarr 
+    import fsspec
+
+Using fsspec to create a pythonic filesystem, provides a convenient way to manage file urls.
+
+.. code:: 
+
+    fs = fsspec.filesystem('s3', anon=True) #S3 file system to manage ERA5 files
+    flist = (fs.glob('s3://era5-pds/2020/*/data/air_pressure_at_mean_sea_level.nc')[:2]
+            + fs.glob('s3://era5-pds/2020/*/data/*sea_surface_temperature.nc')[:2])
+
+    fs2 = fsspec.filesystem('')  #local file system to save final jsons to
+
+    from pathlib import Path
+    import os
+    import ujson
+    
+    so = dict(mode='rb', anon=True, default_fill_cache=False, default_cache_type='first') # args to fs.open()
+    # default_fill_cache=False avoids caching data in between file chunks to lowers memory usage.
+    
+    def gen_json(file):
+        with fs.open(file, **so) as infile:
+            h5chunks = SingleHdf5ToZarr(infile, file, inline_threshold=300) 
+            # inline threshold adjusts the Size below which binary blocks are included directly in the output
+            # a higher inline threshold can result in a larger json file but faster loading time
+            variable = file.split('/')[-1].split('.')[0]
+            month = file.split('/')[2] 
+            outf = f'{month}_{variable}.json' #file name to save json to
+            with fs2.open(outf, 'wb') as f:
+                f.write(ujson.dumps(h5chunks.translate()).encode());
+
+ERA5-pds is located in us-west-2 and so depending on where this computation is taking place the time taken can vary dramatically and is likely to be faster than this.
+
+.. code:: 
+
+    %%time
+    for file in flist:
+        gen_json(file)
+
+
+.. parsed-literal::
+
+    CPU times: user 30.4 s, sys: 4.74 s, total: 35.1 s
+    Wall time: 14min 44s
+
+
+The ``.json`` reference files we have generated can now be used to open virtual datasets through xarray. To achieve this it is first necessary to create a mapping of the reference file using ``fsspec``. Here specifying that the reference json is pointing to files stored on AWS.
+
+.. code:: 
+
+    import xarray as xr
+
+    %%time
+    fs_ = fsspec.filesystem("reference", fo='01_air_pressure_at_mean_sea_level.json', ref_storage_args={'skip_instance_cache':True},
+                           remote_protocol='s3', remote_options={'anon':True})
+    m = fs_.get_mapper("")
+    ds = xr.open_dataset(m, engine="zarr", backend_kwargs={'consolidated':False}, chunks={})
```

Comment:
You can do all this in one line, but its a bit messy:
```
ds = xr.open_dataset("reference://", engine="zarr", backend_kwargs={
    "consolidate": False,
    "storage_options": {"fo": '01_air_pressure_at_mean_sea_level.json', "remote_protocol": "s3",
        "remote_options": {"anon": True}}
)
```
(the complexity of this line is why people would eventually want to hide such a dataset prescription inside an Intake catalogue).